src/events/cd-event-sessions.vue

```diff
@@ -78,22 +67,31 @@
       totalBooked() {
         return this.children.length;
       },
+      applications() {
+        return [].concat(...(this.children).map(child => child.value));
+      },
     },
     methods: {
-      // needs tests for these bois
       async validateAllChildComponents() {
         return Promise.all(this.$refs.allChildComponents.map(
           child => child.$validator.validateAll()));
       },
-      async addChildComponent() {
+      async checkValidatedChildComponents() {
         const validatedChildComponents = await this.validateAllChildComponents();
-        validatedChildComponents.forEach((childIsValid) => {
-          if (!childIsValid) {
-            this.valid = false;
-          } else {
-            this.valid = true;
-          }
-        });
+        if (validatedChildComponents.length > 0) {
+          validatedChildComponents.forEach((childIsValid) => {
```

Comment:
The value of valid is overwritten by the last one. If the last one is valid, everything will be considered as valid.
Removing the if/else and returning `this.valid = this.valid && childIsValid` should be enough.

Reply:
I think `this.valid = childIsValid` is even better bc it uses the latest value :) with the  `this.valid && childIsValid`, if the previous value of this.valid is false (say they forgot their ticket) and then they fix that issue, it will still give a false value (false&&true). thanks

Reply:
Hum, let me rephrase.
In the current scenario, if your first child is invalid but your second child is, this.valid will be true, which is wrong.
The validity upon checking should be resetted at the beginning of this function to ensure that, or I don't get the point of this function.
Technically speaking, this function could be summed up as `validatedChildComp.every((childIsValid) => childIsValid))`

Reply:
Oh ok gotcha!! Will implement the .every in this function